src/components/TodoList.tsx

```diff
@@ -0,0 +1,98 @@
+import { useState } from "react";
+import { useTodoStore, ITodo } from "../useTodoStore";
+import { arrayMove } from "@dnd-kit/sortable";
+import {
+  DndContext,
+  closestCenter,
+  DragEndEvent,
+  DragOverlay,
+  DragStartEvent,
+} from "@dnd-kit/core";
+import Droppable from "./Droppable";
+import TodoItem from "./TodoItem";
+import styled from "styled-components";
+
+const TodoLists = styled.div`
+  display: flex;
+  flex-direction: row;
+`;
+
+function TodoList() {
+  const todos = useTodoStore((state) => state.todos); // 기존 todo list 배열
+  const updateTodos = useTodoStore((state) => state.updateTodos); // todo list 순서 업데이트 함수
+  const updateCategory = useTodoStore((state) => state.updateCategory); // 카테고리 변경 시 todo list 순서 업데이트 함수
```

Comment:
```suggestion
  const { todos, updateTodos, updateCategory } = useTodoStore();
```
zustand 상태와 set 함수는 
디스트럭처링으로 간단하게 가져올 수 있습니다!

Reply:
궁금한게 이렇게 구조분해 할당으로 한번에 호출하면 Zustand에서 스토어의 모든 상태를 가져오는 것 같습니다. 
기존엔 각각 다른 state값을 가져오고있어서 개별적으로 리렌더링이 되었던 반면 한번에 호출하면 하나의 state값만 바뀌어도 useTodoStore 전체가 호출되어 리렌더링이 불필요하게 일어나는 것으로 (gpt와) 파악중인데... 
어떤 방법이 가장 효율적일까요??

Reply:
오 이것은 저도 생각하지 못했던 부분입니다 ㅎㅎ

말씀하신대로 디스트럭처링으로 가져오면 불필요한 리렌더링이 발생하는 문제가 있네요
성능을 고려한다면 기존 방식으로 하는게 맞아보입니다

이런 케이스는 현재 개발중인 앱이 성능과 개발 편의 중 어느게 중요한지 따져보고
적절하게 사용하면 될 것 같습니다!